array-filter/filter.js

```diff
@@ -0,0 +1,7 @@
+function filter(array, predicate) {
+  let result = [];
+  for (ii = 0, ii < array.length, ii++) {
+    const res = predicate(array[ii]);
+    result.push(res);
+  }
```

Comment:
Here you need an `if` statement, `if` the return value from `predicate` is truthy then push the value into the array, if it is falsey you don't push the value into the new array.

Reply:
Whoops! fixed in update :)